CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,494 @@
+                  var collection: RawSyntax
+                  let arena = SyntaxArena()
+                  if let col = raw.layoutView!.children[\(raw: index)] {
+                    collection = col.layoutView!.appending(element.raw, arena: arena)
+                  } else {
+                    collection = RawSyntax.makeLayout(kind: SyntaxKind.\(raw: childNode.swiftSyntaxKind),
+                                                      from: [element.raw], arena: arena)
+                  }
+                  let newData = data.replacingChild(at: \(raw: index), with: collection, arena: arena)
+                  return \(raw: node.name)(newData)
+                }
+                """
+              )
+            }
+          }
+        }
+
+        try! VariableDeclSyntax("public static var structure: SyntaxNodeStructure") {
+          let layout = ArrayExprSyntax {
+            for child in node.children {
+              ArrayElementSyntax(
+                expression: ExprSyntax(#"\Self.\#(raw: child.swiftName)"#)
+              )
+            }
+          }
+
+          StmtSyntax("return .layout(\(layout))")
+        }
+
+        try! FunctionDeclSyntax("public func childNameForDiagnostics(_ index: SyntaxChildrenIndex) -> String?") {
+          try! SwitchStmtSyntax("switch index.data?.indexInParent") {
+            for (index, child) in node.children.enumerated() {
+              SwitchCaseSyntax("case \(raw: index):") {
+                if let nameForDiagnostics = child.nameForDiagnostics {
+                  StmtSyntax(#"return "\#(raw: nameForDiagnostics)""#)
+                } else {
+                  StmtSyntax("return nil")
+                }
+              }
+            }
+
+            SwitchCaseSyntax(
+              """
+              default:
+                fatalError("Invalid index")
+              """
+            )
+          }
+        }
+      }
+
+      try! ExtensionDeclSyntax("extension \(raw: node.name): CustomReflectable") {
+        try! VariableDeclSyntax("public var customMirror: Mirror") {
+          ReturnStmtSyntax(
+            expression: FunctionCallExprSyntax(callee: ExprSyntax("Mirror")) {
+              TupleExprElementSyntax(expression: ExprSyntax("self"))
+              TupleExprElementSyntax(
+                label: "children",
+                expression: DictionaryExprSyntax {
+                  for child in node.children {
+                    DictionaryElementSyntax(
+                      leadingTrivia: .newline,
+                      keyExpression: ExprSyntax(#""\#(raw: child.swiftName)""#),
+                      valueExpression: child.isOptional
+                        ? ExprSyntax("\(raw: child.swiftName).map(Syntax.init)?.asProtocol(SyntaxProtocol.self) as Any")
+                        : ExprSyntax("Syntax(\(raw: child.swiftName)).asProtocol(SyntaxProtocol.self)")
+                    )
+                  }
+                }
```

Comment:
This is nested pretty deep. What do you think about extracting the `DictionaryExprSyntax` into a variable?

```swift
let children = DictionaryExprSyntax { …
```